server/src/test/java/org/opensearch/discovery/InitializeExtensionRequestTest.java

```diff
@@ -0,0 +1,120 @@
+import org.opensearch.extensions.ExtensionDependency;
+import org.opensearch.extensions.ExtensionsManagerTests;
+import org.opensearch.indices.breaker.NoneCircuitBreakerService;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.transport.MockTransportService;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.nio.MockNioTransport;
+
+import java.net.InetAddress;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.emptySet;
+
+public class InitializeExtensionRequestTest extends OpenSearchTestCase {
+
+    private MockTransportService transportService;
+    private MockNioTransport transport;
+    private final ThreadPool threadPool = new TestThreadPool(ExtensionsManagerTests.class.getSimpleName());
+
+    @Before
+    public void setup() throws Exception {
+        Settings settings = Settings.builder().put("cluster.name", "test").build();
+        transport = new MockNioTransport(
+            settings,
+            Version.CURRENT,
+            threadPool,
+            new NetworkService(Collections.emptyList()),
+            PageCacheRecycler.NON_RECYCLING_INSTANCE,
+            new NamedWriteableRegistry(Collections.emptyList()),
+            new NoneCircuitBreakerService()
+        );
+        transportService = new MockTransportService(
+            settings,
+            transport,
+            threadPool,
+            TransportService.NOOP_TRANSPORT_INTERCEPTOR,
+            (boundAddress) -> new DiscoveryNode(
+                "test_node",
+                "test_node",
+                boundAddress.publishAddress(),
+                emptyMap(),
+                emptySet(),
+                Version.CURRENT
+            ),
+            null,
+            Collections.emptySet()
+        );
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        transportService.close();
+        ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+    }
+
+    public void testinitializeExtensionRequest() throws Exception {
+        String expectedUniqueId = "test uniqueid";
+        Version expectedVersion = Version.fromString("2.0.0");
+        ExtensionDependency expectedDependency = new ExtensionDependency(expectedUniqueId, expectedVersion);
+        DiscoveryExtensionNode extensionNode = new DiscoveryExtensionNode(
+            "firstExtension",
+            "uniqueid1",
+            new TransportAddress(InetAddress.getByName("127.0.0.0"), 9300),
+            new HashMap<String, String>(),
+            Version.fromString("3.0.0"),
+            Version.fromString("3.0.0"),
+            List.of(expectedDependency)
+        );
+
+        System.out.println(extensionNode.getName());
```

Comment:
We should not use `System.out` in production code

Reply:
Hi @owaiskazi19 ,

Oops sorry, I must have committed this file by mistake. This is not yet complete and testcase errors out. I have written testcase only for InitializeExtensionResponseTest where I have made the change. I will commit this InitializeExtensionRequestTest separately later.

Thanks